### PR TITLE
Adding SurfaceCtrl and the SurfaceCtrl_demo

### PR DESCRIPTION
**Pull request by Xemuth**
Here is both package which are present in Bazaar. They both rely on Surface package (present on Bazaar) developped by Koldo. 
Feel free to reject/wait this PR untill Surface package has been added to UppHub

### CODE REVIEW

**Comment by klugier** [comment]
@Xemuth excellent contribution! @mirek-fidler would it be possible to add "depends" key in which we will be able to specify which packages/packages (json array) should be downloaded when user click. I think in the second iteration of UppHub (next upp release) we could try to implement versioning for UppHub packages. As I said many times basing on master is difficult - it might work or might not.

Also, SurfaceCtrl_Demo shouldn't be separate UppHub entity. It should be in the same repository as SurfaceCtrl. You should store everything in one repository. Please see how Skylar handle examples in one repo.

@Xemuth can you create forum thread when you ask Koldo to add Surface package to UppHub. For some reason I can not shout him here.

**Review by klugier — CHANGES_REQUESTED** [review]
Let's wait with merging this PR until we will have all pieces.

File: nests.json [hunk 1]
```diff
@@ -54,6 +54,28 @@
...
       "readme": "https://raw.githubusercontent.com/ismail-yilmaz/MessageCtrl/main/README.md"
+    },
+    {
+      "name": "SurfaceCtrl",
+      "packages": [
+        "SurfaceCtrl"
+      ],
+      "description": "A 3D viewer few different 3D object data format",
+      "repository": "https://github.com/Xemuth/SurfaceCtrl.git",
+      "status": "beta",
```
Comment (klugier) [inline nests.json:65]: @ismail-yilmaz could you design which statuses can be set. I think we have experimental which is very similar to beta. We need to create unified protocol.

**Comment by Xemuth** [comment]
@klugier  yes indeed, the Demo should be in same nest. I will see with koldo to add Surface package to UppHub. I have updated the beta to experimental

**Comment by Xemuth** [comment]
I have updated SurfaceCtrl repository to make it look like skylark one (it include demo). I will send message to Koldo about Surface package

File: nests.json (hunk 1 above)
Comment (ismail-yilmaz) [reply nests.json]: @klugier 

Well, the original scheme used in bazaar (the traditional way) was good and familiar enough, IMO. But here we can use:

- `experimental`: Package is new, working but it is not expected to be stable or considered finished.
- `stable`: Package has a stable api and is well tested.
- `rolling`: This means that the package is well tested and stable in the sense that it has a defined behaviour, not crash-prone, but adds new stuff frequently or occasionally (e.g. TerminalCtrl)
- `deprecated`: The package is deprecated, has a better alternative, but possibly required by exisiting apps or other packages.
- `obsolete`: Package is not used anymore. Provided as an archive item.

What do you think? We can change, expand, refine these "scratch" definitions.

**Comment by mirek-fidler** [comment]
I agree. Originally I planned to also have "broken" (not normally listed in
GUI) and maybe something like "malware" (not normally visible).

**Comment by ismail-yilmaz** [comment]
@mirek-fidler 

Adding 'broken' is a good idea too.  I'll add that to the list....

File: nests.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
       "packages": [
         "SurfaceCtrl"
       ],
-      "description": "A 3D viewer few different 3D object data format",
+      "description": "An OpenGL 3D viewer supporting few differents 3D file format",
```
Comment (ismail-yilmaz) [inline nests.json:63]: @Xemuth ,

Just a suggestion: 

"An OpenGL 3D object viewer, supporting various 3D file formats"

Comment (Xemuth) [reply nests.json]: @ismail-yilmaz yes, it's better

File: nests.json (hunk 1 above)
Comment (klugier) [reply nests.json]: @ismail-yilmaz  I am fine with that scheme - please add it to the README.md. Thanks for providing this list.

Comment (ismail-yilmaz) [reply nests.json]: @klugier 

Sure, but I believe he already stated his opinion below (If I am not interpreting wrongly - the 'broken' category is his idea too. ). :)

Comment (klugier) [reply nests.json]: I didn't scroll that far :)

Comment (ismail-yilmaz) [reply nests.json]: @klugier 

No problem.  I am now resolving this conversation and I'll make the relevant changes to the README file. and send a pull request, later tonight.

**Comment by Xemuth** [comment]
how is the PR Going ? Are you waiting for some action from me ? (I'm totally newbie with PR and how it work)

**Comment by ismail-yilmaz** [comment]
@Xemuth 
@klugier 

> how is the PR Going ? Are you waiting for some action from me ? (I'm totally newbie with PR and how it work)

Well, it looks good but AFAIK Surface package is not on upphub yet. Should we wait for it, or the latest TheIDE handles that?

**Comment by ismail-yilmaz** [comment]
Since the required packages are added to the list, I have added SurfaceCtrl as well. See this [PR](https://github.com/ultimatepp/UppHub/pull/14) And I requested review.

Closing this.

**Pull request closed without merging**